pkg/cluster/cluster.go

```diff
@@ -53,9 +54,18 @@ func (c *Cluster) Start(ctx context.Context) (<-chan struct{}, error) {
 			clientURL.Host = clientURL.Hostname() + ":2379"
 			clientURLs = append(clientURLs, clientURL.String())
 		}
-		etcdProxy, err := etcd.NewETCDProxy(ctx, true, c.config.DataDir, clientURLs[0])
-		if err != nil {
-			return nil, err
+
+		t := time.NewTicker(5 * time.Second)
+		defer t.Stop()
+		var etcdProxy etcd.Proxy
+		for range t.C {
+			etcdProxy, err = etcd.NewETCDProxy(ctx, true, c.config.DataDir, clientURLs[0])
+			if err != nil {
```

Comment:
Do we want to only retry on specific errors, such as the port already being in use? Are there other things that might cause this to fail, or is that pretty much it?